### ISSUE TITLE
Issue in the getBusinessObject implementation

### ISSUE THREAD

**Issue by MohamedBakrAli** [bpmn-io/bpmn-js#1764]
__Describe the Bug__

There is an issue in the `getBusinessObject(...)` implementation, This method returns the given `element.businessObject` or the `element` itself if this object does not contain `businessObject` without checking the type of the `element`, I think that not correct.
https://github.com/bpmn-io/bpmn-js/blob/d676f4e962c8058fa5f04fd6aeb12317b98bd227/lib/util/ModelUtil.js#L35-L44

__Steps to Reproduce__

 - Call this method with object `x = {test:"111"}` it will return the object itself.

 __Expected Behavior__

- Return the object itself if the `x` is of type `ModdleElement` or the `x.businessObject` that may be `undefine` like the mentioned example (`x = {test:"111"}`).

__Environment__

 - Browser: [Any]
 - OS: [Any]
 - Library version: [Any]

**Comment by MohamedBakrAli** [comment]
Thanks, @smbea.
I updated it as you mentioned.

**Comment by barmac** [comment]
Does it cause any problems when you use the util in the expected way?

**Comment by MohamedBakrAli** [comment]
Yes, As we build some custom cobonent and utils that deal with the XML, so if we want to read something from the extension element, We should check if the returned from the `getBusinessObject(...)` is a type of `ModdleElement` to avoid break of the code, but if this method return `undefine` in the cases like the mentioned above its handle will so easy and I don't need to see its implementation to check the type of the return is `ModdleElement` or not.

**Comment by barmac** [comment]
Sorry, I don't understand your use case. Can you share a code snippet where the util fails unexpectedly? Note that we assume the passed object is either a diagram-js shape or a ModdleElement.

> Also, it's not correct to return the passed object without checking its type, right?

I'd say it depends on the contract.
